agent/connect/ca/provider.go

```diff
@@ -179,14 +179,17 @@ type SecondaryProvider interface {
 	//
 	// After the certificate is signed, SecondaryProvider.SetIntermediate will
 	// be called to store the intermediate CA.
-	GenerateIntermediateCSR() (string, error)
+	//
+	// The second return value is an opaque string meant to be passed back to
+	// the subsequent call to SetIntermediate.
+	GenerateIntermediateCSR() (string, string, error)
```

Comment:
For the vault provider we needed a way to pass extra data from the `GenerateIntermediateCSR` step to the `SetIntermediate` step so an extra output/input pair was setup to thread an opaque string through.

The vault provider will pass a `keyId`.